system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,98 +28,186 @@ The application package has the following folder structure:
 ```
 
 **Top-level Attributes**
 
 | Attribute       | Type            | Required?       | Description     |
 |-----------------|-----------------|-----------------|-----------------|
-| apiVersion      | string    | Y    | Identifier of the version of the API the object definition follows.|
-| kind            | string    | Y    | Must be `application`.|
-| metadata        | Metadata    | Y    | Metadata element specifying characteristics about the application. See the [Metadata](#metadata-attributes) section below. |
-| spec            | Spec    | Y    | Spec element that defines components of the application. See the [Spec](#spec-attributes) section below. |
+| apiVersion      | string          | Y               | Identifier of the version of the API the object definition follows.|
+| kind            | string          | Y               | Must be `application`.|
+| metadata        | Metadata        | Y               | Metadata element specifying marketing characteristics about the application. See the [Metadata](#metadata-attributes) section below.|
+| components         | Component        | Y               | Component element specifying the software packages to install. See the [Component](#component-attributes) section below. |
 
 **Metadata Attributes**
 
 | Attribute        | Type            | Required?       | Description     |
 |------------------|-----------------|-----------------|-----------------|
-| name             | string          | Y    | The name of the name of the application.|
-| version          | string          | Y    | Version of the application.|
-| description      | string          | Y    | Short application description summary. |
-| release-notes    | string          | N    | Statement about changes of this release of the application.  |
-| icon             | string          | N    | Link to the icon file (e.g., in PNG format).  |
-| author           | string          | N    | Person who created the application. |
-| author-email     | string          | N    | Email address of person who created the application.  |
-| organization     | string          | Y    | Organization that is responsible for the application development. |
-| organization-site| string          | N    | Link to the organization website.  |
-| app-tagline      | string          | N    | A brief statement about the application.  |
-| description-long | string          | N    | A long description about the application.  |
-| license-file     | string          | N    | Link to the file that details the license of the application.|
-| app-site         | string          | N    | Link to the application website. |
-
-**Spec Attributes**
+| id               | string          | Y    | An identifier for the application. The id is used to help create unique identifiers where required, such as namespaces. It must be in reverse domain name notation.|
+| name             | string          | Y    | The application's official name.|
```

Comment:
We should mention, that `name` is a human friendly name including whitespaces and (limited) special characters. `id` should be limited to strings only consisting out of lower case characters, numbers and hyphens/underscores (or any other reasonable specification).